### ISSUE TITLE
Drawing Classifier should not save batch_size

### ISSUE THREAD

**Issue by TobyRoseman** [apple/turicreate#1943]
The drawing classifier saves the `batch_size` values given in the `create(...)` call to its state. It's not clear how/if this is used since `evaluate(...)`, `predict(...)` and `predict_topk(...)` all take a `batch_size` parameter.

In generally we don't want to store batch size in model state. Batch size isn't really model dependent but dependent on the machine it is running on.

**Comment by nickjong** [comment]
The evaluate and predict methods shouldn't take a batch size at all. Let's be consistent with OD and AC for now, which do preserve batch size.

In the general case, batch size can affect model quality, so I think it's legitimate for a user to want to know what the batch size was.

**Comment by TobyRoseman** [comment]
I think we need to have a larger discussion here. I'm surprised that the object detector and activity classifier don't have a `batch_size` parameter on their `evaluate(...)` and `predict(...)` methods. All other toolkits that can use the GPU do have this parameter.

To me the batch size is more about the machine that is running the model than the model itself. Consider the scenario where the model is trained on a powerful machine but the model is used on less powerful machines. I think it's reasonable to want different batch sizes.

**Comment by nickjong** [comment]
What toolkits are you referring to that already expose batch size?

In general, if the question is about machine configuration, we should try to configure the batch size ourselves.

If it's about training behavior (e.g. due to batch norm), then it really only makes sense to configure it on the `create` method.

**Comment by TobyRoseman** [comment]
`image_classifier`, `sound_classifier` and `drawing_classifier` all have a `batch_size` parameter on the following methods: `classify(...)`, `evaluate(...)`, `predict(...)` and `predict_topk(...)`.

`image_similarity` has `batch_size` on `query(...)`. 

`style_transfer` has `batch_size` on `stylize(...)`.

**Comment by nickjong** [comment]
Hmm, for image classification and sound classification, we only perform NN inference, so batch size only determines performance. The object detection, activity classification, and style transfer toolkits (all the toolkits with NN training that predate this year) all store the batch size, which at least in principle affects the quality of the final model by determining how many instances inform each iteration's estimate of the gradient, as well as determining how many updates are performed (when total training time is defined by number of epochs). It's quite reasonable for somebody to try training models varying only the batch size, to evaluate the effect on model quality. For that reason, I think we should store the batch size for any toolkit that updates NN weights.

Consequently, I suggest that inference-based methods default to the stored batch size by default, although certainly it's reasonable for users to override that for performance reasons. The same GPU can likely accommodate larger batch sizes for inference than for training
